src/integrationTest/java/com/mongodb/hibernate/EmbeddableIntegrationTests.java

```diff
@@ -0,0 +1,411 @@
+            item.id = 1;
+            item.omitted = new EmbeddableEmptyValue();
+        }
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find())
+                .containsExactly(
+                        // Hibernate ORM does not store `item.omitted` despite it being non-`null`
+                        new BsonDocument(ID_FIELD_NAME, new BsonInt32(item.id)));
+        var loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithFattenedEmptyValue.class, item.id));
+        // the entity we stored and the entity we loaded are not equal because Hibernate ORM omits `item.omitted`
+        assertNotEquals(item, loadedItem);
+        var updatedItem = sessionFactoryScope.fromTransaction(session -> {
+            var result = session.find(ItemWithFattenedEmptyValue.class, item.id);
+            result.omitted = null;
+            return result;
+        });
+        assertThat(mongoCollection.find()).containsExactly(new BsonDocument(ID_FIELD_NAME, new BsonInt32(item.id)));
+        loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithFattenedEmptyValue.class, item.id));
+        assertEquals(updatedItem, loadedItem);
+    }
+
+    @Test
+    void testNestedValues() {
+        var item = new ItemWithNestedValues();
+        {
+            item.id = new StructAggregateEmbeddableValue();
+            item.id.a = 1;
+            item.nested1 = new StructAggregateEmbeddableValue();
+            item.nested1.a = 2;
+            item.nested2 = new StructAggregateEmbeddablePairValue1();
+            item.nested2.a = 3;
+            item.nested2.nested = new StructAggregateEmbeddablePairValue2(4, 5);
+            item.nested2.parent = item;
+        }
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find())
+                .containsExactly(new BsonDocument()
+                        // Hibernate ORM flattens `item.id` despite it being of an aggregate type
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id.a))
+                        .append("nested1", new BsonDocument("a", new BsonInt32(2)))
+                        .append(
+                                "nested2",
+                                new BsonDocument()
+                                        .append("a", new BsonInt32(3))
+                                        .append(
+                                                "nested",
+                                                new BsonDocument()
+                                                        .append("a", new BsonInt32(4))
+                                                        .append("b", new BsonInt32(5)))));
+        var loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithNestedValues.class, item.id));
+        assertEquals(item, loadedItem);
```

Comment:
I think we should rename this to something like `assertSameValues` - it is not at all equals, but seems to be something much more complex, involving reflection (though it is true that it would ultimately behave like equals under certain circumstances).

Reply:
The current name is intentionally chosen to be what it is. All this method does is saves us from overriding `equals` in the test code.

As for "same" when talking about reference types, I prefer using it only to express [reference equality](https://docs.oracle.com/javase/specs/jls/se24/html/jls-15.html#jls-15.21.3). The Java SE specification attempts to have the distinction between "same" and "equal" in some places, while failing to have it in others 🤷‍♂️

Reply:
`assertEq`?

Reply:
Done in 09d49217b2530d41463e15c8050e25a3e4bd93aa.